java-callout/src/main/java/com/callout/api/Decryption.java

```diff
@@ -71,21 +68,13 @@ public ExecutionResult execute(MessageContext messageContext, ExecutionContext e
                 return ExecutionResult.ABORT;
             }
 
-            // Proceed with decryption
-            Cipher cipherD = Cipher.getInstance("AES/GCM/NoPadding");
-            SecretKeySpec keySpecD = new SecretKeySpec(keyBytes, "AES");
-            GCMParameterSpec gcmParameterSpecD = new GCMParameterSpec(GCM_TAG_LENGTH * 8, ivBytes);
-            cipherD.init(Cipher.DECRYPT_MODE, keySpecD, gcmParameterSpecD);
-            byte[] plainBytesD = cipherD.doFinal(encryptedBytesForDecryption);
-            String originalText = new String(plainBytesD, StandardCharsets.UTF_8);
-            messageContext.setVariable("decrypted_payload", originalText);
+            // Call the decrypt method
+            String decryptedBytes = payloadEncryptorDecryptor.decrypt(Base64.getEncoder().encodeToString(keyBytes), Base64.getEncoder().encodeToString(ivBytes), Base64.getEncoder().encodeToString(encryptedBytesForDecryption));
```

Comment:
No need to encode to string again, pass the key as it is. Key and IV length Check is already handled. Refer: Class:DefaultPayloadEncryptorDecryptor, method: setKeyAndIvInBytes